docs/types/mapping-types.rst

```diff
@@ -1,44 +1,46 @@
 .. index:: !mapping
 .. _mapping-types:
 
-Mapping Types
+Tipos Mapping
 =============
 
-Mapping types use the syntax ``mapping(KeyType => ValueType)`` and variables
-of mapping type are declared using the syntax ``mapping(KeyType => ValueType) VariableName``.
-The ``KeyType`` can be any
-built-in value type, ``bytes``, ``string``, or any contract or enum type. Other user-defined
-or complex types, such as mappings, structs or array types are not allowed.
-``ValueType`` can be any type, including mappings, arrays and structs.
-
-You can think of mappings as `hash tables <https://en.wikipedia.org/wiki/Hash_table>`_, which are virtually initialised
-such that every possible key exists and is mapped to a value whose
-byte-representation is all zeros, a type's :ref:`default value <default-value>`.
-The similarity ends there, the key data is not stored in a
-mapping, only its ``keccak256`` hash is used to look up the value.
-
-Because of this, mappings do not have a length or a concept of a key or
-value being set, and therefore cannot be erased without extra information
-regarding the assigned keys (see :ref:`clearing-mappings`).
-
-Mappings can only have a data location of ``storage`` and thus
-are allowed for state variables, as storage reference types
-in functions, or as parameters for library functions.
-They cannot be used as parameters or return parameters
-of contract functions that are publicly visible.
-These restrictions are also true for arrays and structs that contain mappings.
-
-You can mark state variables of mapping type as ``public`` and Solidity creates a
-:ref:`getter <visibility-and-getters>` for you. The ``KeyType`` becomes a parameter for the getter.
-If ``ValueType`` is a value type or a struct, the getter returns ``ValueType``.
-If ``ValueType`` is an array or a mapping, the getter has one parameter for
-each ``KeyType``, recursively.
-
-In the example below, the ``MappingExample`` contract defines a public ``balances``
-mapping, with the key type an ``address``, and a value type a ``uint``, mapping
-an Ethereum address to an unsigned integer value. As ``uint`` is a value type, the getter
-returns a value that matches the type, which you can see in the ``MappingUser``
-contract that returns the value at the specified address.
+Los tipos mapping usan la sintaxis ``mapping(KeyType => ValueType)`` y las variables
+del tipo mapping son declaradas usando la sintaxis ``mapping(KeyType => ValueType) VariableName``.
+Donde  ``KeyType`` puede ser cualquier tipo estándar, ``bytes``, ``string``, cualquier contrato 
+o un tipo enum. Sin embargo, tipos definidos por el propio usuario o tipos complejos, como 
+mappings, structs o arrays no están permitidos para ``KeyType``. Por otro lado, para ``ValueType``,
+puede ser cualquier tipo, incluyendo mappings, arrays o structs.
+
+Puedes pensar en los mappings como `tablas hash <https://es.wikipedia.org/wiki/Tabla_hash>`_, las
```

Comment:
Since we haven't had many translators we have kind of been sticking to an "unspoken rule" of translating using third person formal "Ud." of "you" in translations. If this is not documented in the README we will need to update it.